### DISCUSSION TITLE
NanoH4 - PCB

### DISCUSSION THREAD

**Discussion by christian-hellwig** [kachurovskiy/nanoels#222] [General]
First thank you for the great project. I wan't to build the H4 but as always there is no reasonable way to get just one manufactured. Since you mentioned you don't want to restock anytime soon, would it be possible that you sell the assembled pcbs via a service like elecrow.com? For example the guy behind FluidNC is selling his hardware there and seems quite happy with it. I ordered some of his 6xCNC Controllers from there and the built quality looks great. P.S. Do you favour any of the alternatives to the mosfets currently used, bc they are EOL? Thanks and happy Sunday

**Comment by kachurovskiy** [comment]
Hi @christian-hellwig,

> you sell the assembled pcbs via a service like elecrow.com

I'm not sure how services like that work but feel free to sell them there yourself, the MIT license for nanoels is incredibly permissive.

> favour any of the alternatives to the mosfets currently used, bc they are EOL?

Pretty much any similar IC would work, there's nothing special about ICs I picked besides them being in the JLCPCB basic parts list.

Thanks!

**Reply by christian-hellwig** [reply]
Thanks for the offer, but I won't feel comfortable selling something I did't develop myself. But I will keep you posted if I manage to build the h4 in time

**Reply by kachurovskiy** [reply]
It would actually be a big service to the community if you could sell them or set them up to be sold by someone.

**Reply by Strongfabandrepair** [reply]
Would it be possible to diy one of these on a cheap proto board using one of those knock off arduino uno boards that uses a esp32? If so could you provide a component list for any other supporting components I would need or that you recommend be included? I know they are shown in the schematic but my ability to follow along written diagrams is lacking unfortunately so even though I know exactly what I am looking at I still don’t fully understand it lol

**Reply by kachurovskiy** [reply]
@Strongfabandrepair I think you can just take a ESP32-S3 dev board like https://www.aliexpress.com/item/1005006370275059.html, keypad reader like https://www.adafruit.com/product/4918 and the LCD2004 display - this combo is enough to wire everything on the breadboard. Many stepper drivers accept 3.3V signals so you won't need any of the level shifters. Buzzer is optional, USB host and the voltage converter are built into the ESP dev board.

**Reply by Strongfabandrepair** [reply]
Thank you for that I’ve been looking for a way to integrate my recently added nema23 x axis power feed into my existing Chinese bought ELS for WM210V metal lathes. Didn’t take long to see no way to modify existing controller and my search for controllers ready made supporting 2 axis became terrifying quickly price wise lol but then I discovered your nanoels videos so truly thank you
